Favor EndpointSlice over Endpoints

sftim opened the pull request:
Split out from https://github.com/kubernetes/website/pull/30817

Document EndpointSlice as the preferred and most appropriate mechanism to record the backing endpoints of a Service.

/sig network

sftim commented:
Helps with https://github.com/kubernetes/website/issues/14770

aojea commented:
/assign @robscott

aojea commented on content/en/docs/concepts/services-networking/service.md:
```diff
@@ -184,73 +189,94 @@ spec:
-subsets:
+  name: my-service-1 # by convention, use the name of the Service
+                     # as a prefix for the name of the EndpointSlice
+  labels:
```
should we suggest to add the other label that indicates the owner?
```
	LabelManagedBy = "endpointslice.kubernetes.io/managed-by"
```

sftim replied:
:thinking: What would we set the value of that label to? The name of the user who is manually adding the EndpointSlice?

sftim replied:
I'm minded to suggest that documenting `endpointslice.kubernetes.io/managed-by` here belongs as a separate change in its own PR.

robscott commented on content/en/examples/access/endpoints-aggregated.yaml:
This is a bit weird in that we accidentally missed adding write access for EndpointSlices to the edit role to begin with. That ended up being a good thing. IIRC, the purpose of this was to allow users to return to the previous "edit" role if desired, but since that never included EndpointSlice write access, I think we should continue to leave that out of this role.

robscott reviewed (commented):
Thanks @sftim! Some great improvements here

sftim commented:
[Changes since last review](https://github.com/kubernetes/website/compare/b12d675d0a52637849d03b2fbf428d28a8582150..379fa060b4d74b0d764d83328a23262fc5927d9f)

Changes from the [second forced push](https://github.com/kubernetes/website/compare/379fa060b4d74b0d764d83328a23262fc5927d9f..7851fb2c7569f8ee612bcf27ecb74c90ff0c590f) update the merge base to be level with _main_, without making other changes.

robscott reviewed (commented):
Thanks @sftim! One nit that is mostly unrelated to your changes but otherwise LGTM.

sftim commented on content/en/docs/concepts/services-networking/service.md:
```diff
@@ -186,73 +191,110 @@ spec:
+
+#### Over-capacity endpoints
+
+If an Endpoints object has more than 1000 endpoints then a Kubernetes v1.22 (or later)
+cluster annotates that Endpoints with `endpoints.kubernetes.io/over-capacity: truncated`.
+This {{< glossary_tooltip text="annotation" term_id="annotation" >}} indicates that the
+affected Endpoints object is over capacity and that the endpoints controller has truncated
+the number of endpoints to 1000.
```
I'll tweak that suggested wording to align with our style guide.

robscott commented:
Thanks @sftim!

/lgtm

sftim commented:
[Changes since previous round of reviews](https://github.com/kubernetes/website/compare/7851fb2c7569f8ee612bcf27ecb74c90ff0c590f..8f4d317cba695e4ac3ff0748850190ebd5d962d8)

reylejano commented:
/label refactor
/approve

sftim commented:
Not sure that https://github.com/kubernetes/website/pull/36673#issuecomment-1284208158 worked
/approve
